### PR TITLE
Abstract Confirmation dialog and update Dialog Button

### PR DESCRIPTION
**Pull request by SferaDev**
### :memo: Implementation

- Abstract Confirmation Dialog from metadata-sync

- Improve the code for generic usage

- Update Dialog button to use Confirmation Dialog as discussed on #8

### CODE REVIEW

File: src/confirmation-dialog/ConfirmationDialog.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            ...other
+        } = this.props;
+
+        return (
+            <React.Fragment>
+                <Dialog open={isOpen} onClose={onCancel} {...other}>
+                    <DialogTitle>{title}</DialogTitle>
+
+                    <DialogContent>
+                        {description && { description }}
```
Comment (tokland) [inline src/confirmation-dialog/ConfirmationDialog.jsx]: I get an error on this: `Objects are not valid as a React child (found: object with keys {description})`. Maybe `description ? description : null`, or even `description || null`. (`null` is important, as we know, React does not accept `undefined` as node)

File: src/confirmation-dialog/ConfirmationDialog.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            onCancel,
+            isOpen,
+            children,
+            cancelText,
+            saveText,
+            ...other
+        } = this.props;
+
+        return (
+            <React.Fragment>
```
Comment (tokland) [inline src/confirmation-dialog/ConfirmationDialog.jsx]: Can we remove the React.Fragment, as we only have one child?

File: src/index.ts [hunk 3]
```diff
@@ -17,7 +18,8 @@ export {
     MultiSelector,
     SimpleCheckBox,
     SearchBox,
-    DialogHandler,
+    DialogButton,
+    ConfirmationDialog,
```
Comment (tokland) [inline src/index.ts:22]: Since we are exposing ConfirmationDialog, should we an entry in the README?

**Review by tokland — COMMENTED** [review]
I added a rule for eslint, for _ignoreThisVar to work. Also some comments:

File: src/confirmation-dialog/ConfirmationDialog.jsx (hunk 1 above)
Comment (SferaDev) [reply src/confirmation-dialog/ConfirmationDialog.jsx]: I finally added something like this:

```js
{description && <DialogContentText>{description}</DialogContentText>}
```

**Review by tokland — APPROVED** [review]
All good to me!

**Pull request merged**